### PR TITLE
Add sitemap.

### PR DESCRIPTION
**Pull request by jsimpson-cc**
[#147443435]

### CODE REVIEW

**Comment by jsimpson-cc** [comment]
Do we want /search in here?

**Comment by scytacki** [comment]
I don't think we need /search it is really just for advanced users.

File: app/controllers/robots_controller.rb [hunk 1]
```diff
@@ -45,4 +54,94 @@ def index
...
+                    xml.url {
+                        xml.loc "#{APP_CONFIG[:site_url]}/resources/#{filter}/#{value}"
+                    }
+                end
+            end
+
+            #
+            # Add activities, sequences and interactives.
+            #
+            materials = ExternalActivity.all + Interactive.all
```
Comment (scytacki) [inline app/controllers/robots_controller.rb]: We don't want all materials here. It should only be the public materials.

**Comment by scytacki** [comment]
One more condition on this, since we just discovered it in the related resources.  If the material is marked as an 'assessment_item'  it should also not be shown in the sitemap. 

I think it would make sense to put all of this logic into a new method in the material_shared_policy.rb which is included in ExternalActivity and Interactive.  Then this policy method can be used both here and in the `home_controller#stem_resources` code.  I was just looking at that `stem_resources` action it doesn't appear take into account the assessment_item or the cohorts, but it should. So putting this in a policy method would unify at least these two parts.  The Solr filtering doesn't fit in the normal policy approach but I think we could figure out a way to add it in.

**Comment by scytacki** [comment]
Perhaps this new method is `visible?`, that would mimic the `ProjectPolicy#visible?` method. 

And to clarify, I don't think the Solr filtering needs to be moved into the policy now. I was just musing about a future improvement.

**Comment by scytacki** [comment]
This is missing the change to `home_controller#stem_resources` but that can be done separately.

**Pull request merged**